### PR TITLE
Migrate csl_item_set_standard_id to CSL_Item class

### PR DESCRIPTION
**Pull request by dhimmel**
Migrate additional functionality to the CSL_Item class. Specifically, csl_item_set_standard_id is now CSL_Item.starndardize_id. append_to_csl_item_note is now note_append_dict and note_append_text

### CODE REVIEW

**Comment by epogrebnyak** [comment]
Is this the branch where you work on add note/read note mechanism? Consider the code blow to refactor 

```
    # append_to_csl_item_note() can be implemented using two methods below.


    def append_note_text(self, text: str):
        note = str(self.get('note', ''))
        if note and not note.endswith('\n'):
            note += '\n'
        note += text
        self['note'] = note
    
    def append_note_dict(self, dictionary: dict):
        def log(key, reason: str):
            msg = (f'append_to_csl_item_note: skipped adding "{key}",'
                     '\nreason: {reason}')
            logging.warning(msg)            
        dict_items = []                            
        for key, value in dictionary.items():
            if not re.fullmatch(r'[A-Z]+|[-_a-z]+', key):
                log(key, 'value "{value}" does not conform to the variable_name ' 
                         'syntax as per https://git.io/fjTzW.')                
                continue
            if '\n' in value:
                log(key, 'value "{value}" contains a newline')
                continue
            dict_items.append(f'{key}: {value}')
        self.append_note_text('\n'.join(dict_items))    


    #     note_text = f'This CSL JSON Item was automatically generated by Manubot v{manubot_version} using citation-by-identifier.'
    #     note_dict = {
    #         'standard_id': citekey,
    #     }
    #     append_to_csl_item_note(csl_item, note_text, note_dict)


    def add_note_manubot_version(self, version=None):
        if version is None:
            from manubot import __version__ as manubot_version
            version = manubot_version
        self.append_note_text('This CSL JSON Item was automatically generated '
                              f'by Manubot v{version} using citation-by-identifier.')
        return self


    def add_note_standard_id(self, citekey:str):
        self.append_note_dict({'standard_id': citekey})
        return self
```

https://github.com/epogrebnyak/manubot/blob/b74a6a71cff97c45a1f2da7c533261104123931c/manubot/cite/csl_item.py#L97-L139

**Comment by dhimmel** [comment]
> Consider the code below to refactor

Nice ideas. I incorporated some parts.

File: manubot/cite/csl_item.py [hunk 1]
```diff
@@ -145,64 +145,121 @@ def clean(self, prune: bool = True):
...
-def assert_csl_item_type(x):
-    if not isinstance(x, CSL_Item):
-        raise TypeError(
-            f'Expected CSL_Item object, got {type(x)}')
+    @note.setter
+    def note(self, text: str):
+        if text:
+            self['note'] = text
+        else:
+            self.pop('note', None)
```
Comment (dhimmel) [inline manubot/cite/csl_item.py]: @epogrebnyak I haven't used a property setter before, but I think this is right.

Comment (epogrebnyak) [reply manubot/cite/csl_item.py]: Niether did I, just learnt recently. What is the behavior idea behind else?

Comment (dhimmel) [reply manubot/cite/csl_item.py]: Added a comment in https://github.com/manubot/manubot/pull/157/commits/97df68a905e531ca0dcdfe85c9d67a3853eb2516 to make this more clear:

> if text is None or an empty string, remove the "note" field

**Pull request merged**